test/Remittance.js

```diff
@@ -0,0 +1,339 @@
+        });
+    
+        it('Should fail if the address is invalid', async () => {
+            await truffleAssert.reverts(
+                instance.newTransaction(nil, hashedOTP, fiveDays, {from: owner, value: toWei("0.1", "ether")})
+            );       
+        });
+
+        it('Should fail if the hashOTP is invalid', async () => {
+            await truffleAssert.reverts(
+                instance.newTransaction(exchange, web3.utils.utf8ToHex(""), fiveDays, {from: owner, value: toWei("0.1", "ether")})
+            );          
+        });
+
+        it('Should fail if the tx value is equal to 0', async () => {
+            await truffleAssert.reverts(
+                instance.newTransaction(exchange, password, fiveDays, {from: owner})
+            );          
+        });
+
+        it('Should fail if the hashedOTP is already used', async () => {
+            await instance.newTransaction(exchange, hashedOTP, fiveDays, {from: owner, value: toWei("0.1", "ether")})
+            await truffleAssert.reverts(
+                instance.newTransaction(exchange, hashedOTP, fiveDays, {from: owner, value: toWei("0.1", "ether")})
+            );
+        });
+    })
+
+    describe('======= cancelTransaction unit testing =======', () => {
+        let hashedOTP;
+
+        beforeEach('Creating new transaction', async () => {
+            hashedOTP = await instance.hashOTP(exchange, password);
+
+            const txObj = await instance.newTransaction(exchange, hashedOTP, fiveDays, {from: owner, value: toWei("0.1", "ether")});
+
+            assert.strictEqual(txObj.logs.length, 2);
+            assert.strictEqual(txObj.logs[0].event, "LogTakeFees");
+            assert.strictEqual(txObj.logs[0].args[0], owner);
+            assert.strictEqual(txObj.logs[0].args[1].toString(), "2000");
+            assert.strictEqual(txObj.logs[1].event, "LogNewTransaction");
+            assert.strictEqual(txObj.logs[1].args[0], owner);
+            assert.strictEqual(txObj.logs[1].args[1], exchange);
+            assert.strictEqual(txObj.logs[1].args[2].toString(), new BN(toWei("0.1", "ether")).minus(2000).toString());
+        });
+
+        it('Should be able to cancel a transaction after one 1 day time', async () => {
+            hashedOTP = await instance.hashOTP(owner, password);
+            await instance.newTransaction(owner, hashedOTP, 1 * 24 * 60 * 60, {from: owner, value: toWei("0.1", "ether")});
+
+            await increaseTime(1 * 24 * 60 * 60);
+    
+            let txObj = await instance.cancelTransaction(owner, hashedOTP, {from: owner});
+            assert.strictEqual(txObj.logs.length, 1);
+            assert.strictEqual(txObj.logs[0].event, "LogCancelTransaction");
+            assert.strictEqual(txObj.logs[0].args[0], owner);
+            assert.strictEqual(txObj.logs[0].args[1], owner);
+            assert.strictEqual(txObj.logs[0].args[2].toString(), new BN(toWei("0.1", "ether")).minus(2000).toString());
```

Comment:
What about `toWei(new BN("0.1"), "ether")`

Reply:
Cause I need to perform operation. Moreover, your way of doing it is triggering an error. Waht is wrong by using this method? Why should I changed it?

Reply:
I was hoping that my syntax would be clearer. It ends in `)` instead of `))`. :)